distribution/lib/Standard/Base/0.0.0-dev/src/System/File_Format.enso

```diff
@@ -118,12 +123,50 @@ type File_Format
         _ = [stream, metadata]
         Unimplemented.throw "This is an interface only."
 
+    ## PRIVATE
+       A static method on each format that returns a vector of options that can
+       be displayed in format selectors that allow choosing this file format.
+
+       A single format instance can provide multiple options to choose, or none
+       at all.
+    get_dropdown_options : Vector Option
+    get_dropdown_options = Unimplemented.throw "This is an interface only."
+
+    ## PRIVATE
+       A static method on each format that returns a vector of name pattern
+       options that can be displayed in the `name_filter_widget`.
+    get_name_patterns -> Vector File_Name_Pattern = Unimplemented.throw "This is an interface only."
+
+    ## PRIVATE
+       Returns a list of all name patterns of all known file formats.
+    all_known_name_patterns -> Vector Text =
+        format_types.flat_map .get_name_patterns . flat_map .patterns . distinct
+
     ## PRIVATE
     default_widget : Widget
     default_widget =
         options = ([Auto_Detect]+format_types).flat_map .get_dropdown_options
         Single_Choice display=Display.Always values=options
 
+    ## PRIVATE
+       Builds a widget intended to be used for `name_filter` of `File.list` and
+       its siblings that allows to filter file names by file format.
+    name_filter_widget -> Widget =
+        known_patterns = File_Format.all.flat_map .get_name_patterns
```

Comment:
Does `File_Format.all` still include `Auto_Detect`?

Reply:
Yes:
```
all = [Auto_Detect] + format_types
```

Reply:
Ok, I had missed that you added `get_name_patterns` to that.

Reply:
Yep, I thought that since `Auto_Detect` will accept any file from the known formats, we could define its set of patterns as a union of known formats' patterns. This also comes in handy because we wanted the "All known formats" entry in the dropdown.